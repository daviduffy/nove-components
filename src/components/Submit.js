// External Dependencies
import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'glamor';

const Submit = ({ label, style }) => (
  <div className="F__g F__g--submit">
    <button className="B B--p F__sb" {...css(style)}>{label}</button>
  </div>
);
Submit.propTypes = {
  label: PropTypes.string,
  style: PropTypes.object.isRequired
};
Submit.defaultProps = {
  label: 'Submit'
};
export default Submit;
