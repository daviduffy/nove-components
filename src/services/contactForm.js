import React from 'react';
import PropTypes from 'prop-types';

import Drawer from '../components/Drawer';
import InputDate from '../components/InputDate';
import InputText from '../components/InputText';
import InputSelect from '../components/InputSelect';
import InputTextarea from '../components/InputTextarea';
import Row from '../components/Row';
import Submit from '../components/Submit';

export const getRenderedInputs = ({
  accordionOpen,
  onAccordionClick,
  inputComponents,
  order,
  style
}) => {
  const renderInput = (({ id, items, ...rest }, i) => {
    if (items) {
      let config;
      let Component;
      switch (id) {
        case 'drawer':
          Component = Drawer;
          config = ({
            ...rest,
            key: `${id}-${i}`,
            isOpen: accordionOpen,
            items: items.map(renderInput),
            onClick: onAccordionClick
          });
          break;
        case 'row':
          Component = Row;
          config = ({
            ...rest,
            key: `${id}-${i}`,
            items: items.map(renderInput)
          });
          break;
        default:
          break;
      }
      return (<Component {...config} />);
    }
    if (id === 'submit') return (<Submit key={i} style={style.submit} {...rest} />);
    const { Component, ...REST } = inputComponents.find(({ id: arrayID }) => arrayID === id);
    return (
      <Component key={i} {...REST} />
    );
  });
  renderInput.propTypes = {
    id: PropTypes.string.isRequired,
    items: PropTypes.oneOfType([
      PropTypes.bool,
      PropTypes.array
    ])
  };
  renderInput.defaultProps = { items: false };
  return order.map(renderInput);
};

export const getInputComponents = (props) => {
  const getInputElement = ({
    hidden = false,
    id,
    label,
    options,
    placeholder,
    required,
    type,
    value
  }) => {
    let dynamicOnChange = props.onInputChange;
    let dynamicOnBlur = () => {};
    const config = {
      Component: InputText,
      label,
      id,
      placeholder,
      value,
      required
    };

    switch (type) {
      case 'date':
        return ({
          ...config,
          Component: InputDate,
          error: (required && props.formError && props.eventDate === undefined),
          onDateChange: props.onDateChange,
          value: props.eventDate,
          windowWidth: props.windowWidth
        });
      case 'select':
        config.Component = InputSelect;
        config.options = options ? [['', placeholder], ...options] : false;
        break;
      case 'textarea':
        config.Component = InputTextarea;
        break;
      case 'phone':
        dynamicOnBlur = props.onPhoneBlur;
        break;
      case 'email':
        dynamicOnBlur = props.onEmailBlur;
        break;
      case 'currency':
        dynamicOnChange = props.onAmountChange;
        break;
      default:
        break;
    }

    return ({
      ...config,
      classes: `F__g--${id}`,
      error: required && props.formError !== '' && value === '',
      onChange: dynamicOnChange,
      onBlur: dynamicOnBlur,
      style: ({ ...(hidden ? { display: 'none' } : {}) }),
      required
    });
  };
  return props.inputs.map(getInputElement);
};
