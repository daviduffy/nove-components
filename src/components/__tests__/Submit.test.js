// External Dependencies
import React from 'react';
import { shallow } from 'enzyme';
import { FORM_INPUTS_DEFAULT } from 'nove-common';

// Internal Dependencies
import Submit from '../Submit';
import { customSubmitStyle } from '../../fixtures/forms';
import { getCSS, getStyles } from '../../services/contactForm';

// not testing 'remote config' because fetch is performed by SignupRetriever
let wrapper;
const props = {
  style: getCSS(getStyles(customSubmitStyle)),
  ...FORM_INPUTS_DEFAULT.submit
};

// console.log(props);

beforeEach(() => {
  wrapper = shallow(<Submit {...props} />);
});

// submit
// =================================================================================================
test('should display normal submit label and placeholder', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should display custom submit text', () => {
  const newProps = { ...props, label: 'go' };
  wrapper = shallow(<Submit {...newProps} />);
  expect(wrapper.find('.F__sb').text()).toBe(newProps.label);
});
