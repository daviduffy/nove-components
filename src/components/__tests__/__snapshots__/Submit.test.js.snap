// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display normal submit label and placeholder 1`] = `
<div
  className="F__g F__g--submit"
>
  <button
    className="B B--p F__sb"
    data-css-14f88ev=""
  >
    Submit
  </button>
</div>
`;
